src/clr-angular/utils/drag-and-drop/draggable.ts

```diff
@@ -3,24 +3,27 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {Directive, ElementRef, OnDestroy, OnInit} from "@angular/core";
+import {AfterContentInit, Directive, ElementRef, OnDestroy} from "@angular/core";
 import {Subscription} from "rxjs/Subscription";
-
 import {ClrDragEvent} from "./interfaces/drag-event";
 import {ClrDragEventListener} from "./providers/drag-event-listener";
+import {ClrDragHandleRegistrar} from "./providers/drag-handle-registrar";
+
 
-@Directive({selector: "[clrDraggable]", providers: [ClrDragEventListener], host: {class: "draggable"}})
-export class ClrDraggable<T> implements OnInit, OnDestroy {
+@Directive(
+    {selector: "[clrDraggable]", providers: [ClrDragEventListener, ClrDragHandleRegistrar], host: {class: "draggable"}})
+export class ClrDraggable<T> implements AfterContentInit, OnDestroy {
     private draggableEl: Node;
 
     private subscriptions: Subscription[] = [];
 
-    constructor(private el: ElementRef, private dragEventListener: ClrDragEventListener<T>) {
+    constructor(private el: ElementRef, private dragEventListener: ClrDragEventListener<T>,
+                private dragHandleRegistrar: ClrDragHandleRegistrar<T>) {
         this.draggableEl = this.el.nativeElement;
     }
 
-    ngOnInit() {
-        this.dragEventListener.attachDragListeners(this.draggableEl);
+    ngAfterContentInit() {
```

Comment:
Why delay from `ngOnInit` to `ngAfterContentInit`?

Reply:
During `ngOnInit`, we don't know if there will be a nested `clrDragHandle` or not. So this is what happens there:
1. Register draggable element as the default drag handle and make it the active drag handle anyways.
2. if `clrDragHandle` is registered sometime later, override the current active drag handle and make the `clrDragHandle`'s element the active drag handle.

But during `ngOnAfterInit`, we know whether we have nested elements or not. And if one of those nested elements happen to be `clrDragHandle` element, make that the active drag handle. If there is no nested `clrDragHandle`, just make the `clrDraggable` element the active drag handle. So basically, using `ngOnAfterInit` prevents `clrDraggable` element from being registered too early and overriden later.